tests/deque.rs

```diff
@@ -0,0 +1,158 @@
+use std::cell::RefCell;
+use std::collections::VecDeque;
+use std::rc::Rc;
+
+use left_right::{
+    aliasing::{Aliased, DropBehavior},
+    Absorb, ReadHandle,
+};
+
+// Value encapsulates an integer value and keeps a registry of live values up to
+// date.
+struct Value {
+    v: i32,
+    r: Rc<ValueRegistry>,
+}
+
+impl Value {
+    fn new(v: i32, r: Rc<ValueRegistry>) -> Self {
+        r.adjust_count(1);
+        Self { v, r }
+    }
+}
+
+impl Drop for Value {
+    fn drop(&mut self) {
+        self.r.adjust_count(-1);
+    }
+}
+
+// ValueRegistry keeps track of the number of Values that have been created and
+// not yet dropped.
+struct ValueRegistry {
+    num_live_values: RefCell<i64>,
+}
+
+impl ValueRegistry {
+    fn new() -> Self {
+        Self {
+            num_live_values: RefCell::new(0),
+        }
+    }
+
+    fn adjust_count(&self, delta: i64) {
+        let mut live_vals = self.num_live_values.borrow_mut();
+        *live_vals += delta;
+        assert!(*live_vals >= 0);
+    }
+
+    fn expect(&self, expected_count: i64) {
+        assert_eq!(*self.num_live_values.borrow(), expected_count);
+    }
+}
+
+struct NoDrop;
+impl DropBehavior for NoDrop {
+    const DO_DROP: bool = false;
+}
+
+struct DoDrop;
+impl DropBehavior for DoDrop {
+    const DO_DROP: bool = true;
+}
+type Deque = VecDeque<Aliased<Value, NoDrop>>;
+
+enum Op {
+    PushBack(Aliased<Value, NoDrop>),
+    PopFront,
+}
+
+impl Absorb<Op> for Deque {
+    fn absorb_first(&mut self, operation: &mut Op, _other: &Self) {
+        match operation {
+            Op::PushBack(value) => {
+                self.push_back(unsafe { value.alias() });
+            }
+            Op::PopFront => {
+                self.pop_front();
+            }
+        }
+    }
+
+    fn absorb_second(&mut self, operation: Op, _other: &Self) {
+        let with_drop: &mut VecDeque<Aliased<Value, DoDrop>> =
```

Comment:
I'd like to see a comment here explaining why this is safe (I agree that it is).